SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -332,24 +332,50 @@ QueryConstraint InterestFactory::CreateCheckoutRadiusConstraints() const
 	DefaultCheckoutRadiusConstraint.RelativeCylinderConstraint = RelativeCylinderConstraint{ DefaultCheckoutRadiusMeters };
 	CheckoutRadiusConstraints.OrConstraint.Add(DefaultCheckoutRadiusConstraint);
 
-	// For every interest distance that we still want, add a constraint with the distance for the actor type and all of its derived types.
-	for (const auto& InterestDistanceSquared: ClientInterestDistancesSquared)
+	TMap<float, TArray<UClass*>> DistanceToActorTypes;
```

Comment:
nit: we can just do this once when the ClientInterestDistancesSquared map is created

Reply:
this is more of a serious flaw than a nit, will fix

Reply:
done, made the entire checkout radius constraint building flow static and will be re-initialised when gather client interests would previously have been called
means only the constraint is stored for re-use later

Reply:
also added comment mentioning when the constraint gets built